python/pyspark/sql/context.py

```diff
@@ -204,15 +206,31 @@ def registerFunction(self, name, f, returnType=StringType()):
         >>> sqlContext.sql("SELECT stringLengthInt('test')").collect()
         [Row(stringLengthInt(test)=4)]
 
+        >>> from pyspark.sql.types import IntegerType
+        >>> from pyspark.sql.functions import udf
+        >>> slen = udf(lambda s: len(s), IntegerType())
+        >>> _ = sqlContext.udf.register("slen", slen)
+        >>> sqlContext.sql("SELECT slen('test')").collect()
+        [Row(slen(test)=4)]
+
         >>> import random
         >>> from pyspark.sql.functions import udf
-        >>> from pyspark.sql.types import IntegerType, StringType
+        >>> from pyspark.sql.types import IntegerType
         >>> random_udf = udf(lambda: random.randint(0, 100), IntegerType()).asNondeterministic()
-        >>> newRandom_udf = sqlContext.registerFunction("random_udf", random_udf, StringType())
+        >>> newRandom_udf = sqlContext.udf.register("random_udf", random_udf)
```

Comment:
I know it's a bit confusing. It's because we started to have the same names in the API. Similar things also apply to R. We follow PEP 8 with few exceptions. It should be with underscore in general if possible. There's an example to refer , `threading.py` in Python. It also happened to have the similar case with us.

Reply:
I am fine about the naming convention. Do we have a style recommendation for it?

Reply:
What do you mean by a style recommendation?

Reply:
Ah, do you maybe literally mean like Scala style guide? It's basically PEP 8.

Reply:
http://spark.apache.org/contributing.html I just checked it. It is already documented.